library/modules/bigip_device_auth_radius.py

```diff
@@ -0,0 +1,765 @@
+
+        if resp.status not in [200, 404]:
+            raise F5ModuleError(resp.content)
+
+        uri = "https://{0}:{1}/mgmt/tm/auth/radius-server/~Common~system_auth_name2".format(
+            self.client.provider['server'],
+            self.client.provider['server_port']
+        )
+        resp = self.client.api.delete(uri)
+
+        if resp.status not in [200, 404]:
+            raise F5ModuleError(resp.content)
+
+        return True
+
+    def read_current_radius_config(self):
+        uri = "https://{0}:{1}/mgmt/tm/auth/radius/~Common~system-auth".format(
+            self.client.provider['server'],
+            self.client.provider['server_port']
+        )
+
+        resp = self.client.api.get(uri)
+
+        try:
+            response = resp.json()
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
+
+        if resp.status in [200, 201] or 'code' in response and response['code'] in [200, 201]:
+            return response
+        raise F5ModuleError(resp.content)
+
+    def read_current_radius_servers(self):
+        servers = list()
+        for name in ['system_auth_name1', 'system_auth_name2']:
+            uri = "https://{0}:{1}/mgmt/tm/auth/radius-server/~Common~{2}".format(
+                self.client.provider['server'],
+                self.client.provider['server_port'],
+                name
+            )
+            resp = self.client.api.get(uri)
+            try:
+                response = resp.json()
+            except ValueError as ex:
+                raise F5ModuleError(str(ex))
+
+            if resp.status not in [200, 201] or 'code' in response and response['code'] not in [200, 201]:
+                if (resp.status in [404] or 'code' in response and response['code'] not in [404]) and name == 'system_auth_name1':
+                    raise F5ModuleError(str(ex))
+            else:
+                servers.append(response)
+
+        return servers
+
+    def read_current_from_device(self):
+        result = dict()
+        curr_config = self.read_current_radius_config()
+
+        if 'serviceType' in curr_config:
+            result['service_type'] = curr_config['serviceType']
+        if 'retries' in curr_config:
+            result['retries'] = curr_config['retries']
+        if 'accountingBug' in curr_config:
+            result['accounting_bug'] = curr_config['accountingBug']
+
+        result['servers'] = self.read_current_radius_servers()
+        result['auth_source'] = self.read_current_auth_source_from_device()
+        return ApiParameters(params=result)
+
+
+class ArgumentSpec(object):
+    def __init__(self):
+        self.supports_check_mode = True
+        argument_spec = dict(
+            retries=dict(
+                type='int'
+            ),
+            service_type=dict(
+                choices=[
+                    'authenticate-only', 'login', 'default', 'framed', 'callback-login', 'callback-framed', 'outbound', 'administrative',
```

Comment:
make this list vertical its easier to read

Reply:
This comment is addressed.